File: javascript/gen_strings.js
// Generate all combinations of strings with specified maximum
// lenght and set of characters to use

function gen_strings(str_length, charset) {
    charset = charset.split('');
    strings = charset

    for ( let i = 1; i < str_length; i++ ) {
        let temp = [];
        for ( let k = 0; k < strings.length; k++ ) {
            string = strings[k];
            if (i === 1) {
                temp.push(string);
            }
            for ( let m = 0; m < charset.length; m++ ) {
                let letter = charset[m];
                temp.push(string + letter);
            }
        }
        strings = temp;
    }
    return strings;
}

File: python/genStrings.py
# Generate all combinations of strings with specified maximum
# lenght and set of characters to use

def gen_strings(length, charset):
    charset = [*charset]
    strings = charset

    for i in range(1, length):
        temp = []
        for k in range(len(strings)):
            string = strings[k]
            if i == 1:
                temp.append(string)
            for m in range(len(charset)):
                letter = charset[m]
                temp.append(string + letter)
        strings = temp
    return(strings)